src/battle_game/core/units/magic_tank.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include "battle_game/core/unit.h"
+#include "battle_game/core/units/tiny_tank.h"
+namespace battle_game::unit {
+class MagicTank : public Tank {
+ public:
+  MagicTank(GameCore *game_core, uint32_t id, uint32_t player_id);
+  void Render() override;
+  void Update() override;
+  uint32_t protectskill_countdown_ = {0};
+  uint32_t protectcool_countdown = 60;
+  uint32_t deathcool_countdown_ = 0;
+  bool protecteed = false;
+  void Protect();
+  void Death();
+  void ProtectClick();
+  void DeathClick();
+  void magic();
```

Comment:
Why use public variables and functions instead of protected/private? u should be really careful and have strong reasons when setting them public. (And ur naming is not consistent, vairables in public can just look like `deathcool_coutdown` and functions should look like`Magic()`)

Reply:
Get it. L change it right now.